### ISSUE TITLE
Preline Dropdown only works after hard refresh

### ISSUE THREAD

**Issue by insivika** [htmlstreamofficial/preline#301]
Just implemented the premium dashboard component in my next.js project and came to find that i have to refresh the page to get the drop down to show up. 

Clicking the profile button doesnt do anything:
![Screenshot 2024-03-11 at 3 41 19 PM](https://github.com/htmlstreamofficial/preline/assets/33808429/5f18467d-d8ca-4f38-adce-6b4f29f4a1fe)

After refreshing the page clicking the button opens the modal:
![Screenshot 2024-03-11 at 3 42 52 PM](https://github.com/htmlstreamofficial/preline/assets/33808429/78339d9e-5fa5-4f7c-854f-932f94dbeac5)

Here's my `PrelineScript` 

```

'use client';

import { usePathname } from 'next/navigation';
import { useEffect } from 'react';

import { IStaticMethods } from 'preline/preline';
declare global {
  interface Window {
    HSStaticMethods: IStaticMethods;
  }
}

export default function PrelineScript() {
  const path = usePathname();

  useEffect(() => {
    const loadPreline = async () => {
      await import('preline/preline');
      window.HSStaticMethods.autoInit();
    };
    loadPreline();
  }, [path]);

  return null;
}


```

**Comment by amrajat** [comment]
Same issue I am experiencing, the advanced select works only after a hard refresh on the first render it looks something like this.

![image](https://github.com/htmlstreamofficial/preline/assets/26631696/68898e81-a8c8-44f9-8b82-ea0d192648fe)

**Comment by insivika** [comment]
I noticed a different number of preline event listeners that get added to the window when I navigate to the page vs when I hard refresh
Navigate to the page:
![Screenshot 2024-03-12 at 11 33 42 AM](https://github.com/htmlstreamofficial/preline/assets/33808429/178aabe5-34c7-454b-8ddf-13a72fddcf13)
Hard refresh:
![Screenshot 2024-03-12 at 11 34 02 AM](https://github.com/htmlstreamofficial/preline/assets/33808429/284b0987-0ec3-46fe-a4ec-ae0645b99814)

**Comment by insivika** [comment]
I see that the click event listener is not added to the button when navigating to the page, only when hard refreshing

**Comment by algebra2boy** [comment]
I have this issue as well. I have to hard refreshing my page to get the popover. Did you guys find a solution to solve this?

**Comment by insivika** [comment]
I had to change the PrelineScript back to how its suggested in the docs and increase the setTimeout delay to 1000

```
'use client';

/* eslint-disable @typescript-eslint/no-explicit-any */
import { usePathname } from 'next/navigation';
import { useEffect } from 'react';

declare global {
  interface Window {
    HSStaticMethods: any;
  }
}

export default function PrelineScript() {
  const path = usePathname();
  useEffect(() => {
    import('preline/preline');
  }, []);

  useEffect(() => {
    setTimeout(() => {
      window.HSStaticMethods.autoInit();
    }, 1000);
  }, [path]);

  return null;
}
'use client';

/* eslint-disable @typescript-eslint/no-explicit-any */
import { usePathname } from 'next/navigation';
import { useEffect } from 'react';

declare global {
  interface Window {
    HSStaticMethods: any;
  }
}

export default function PrelineScript() {
  const path = usePathname();
  useEffect(() => {
    import('preline/preline');
  }, []);

  useEffect(() => {
    setTimeout(() => {
      window.HSStaticMethods.autoInit();
    }, 1000);
  }, [path]);

  return null;
}
```

**Comment by algebra2boy** [comment]
Hey, thanks for the quick reply. The suggested 1000 setTime delay works on my end! Thanks again.

**Comment by amrajat** [comment]
after changing the delay to 1000, it works for rendering only but its functionality still not working. 

```window.addEventListener("load", (event) => {
    const planEl = HSSelect.getInstance("#plan");
    planEl.on("change", (val) => setPlan(Number(val)));

    const planTypeEl = HSSelect.getInstance("#planType");
    planTypeEl.on("change", (val) => setPlanType(val));

    const quantityEl = HSInputNumber.getInstance("#quantity");
    quantityEl.on("change", ({ inputValue }) =>
      setQuantity(Number(inputValue))
    );
  });
```
When we try to select options it doesn't work.

**Comment by landrybielefeld** [comment]
Added and working
![image](https://github.com/htmlstreamofficial/preline/assets/68716899/79229e91-659e-4d6f-a941-885aff677db2)

**Comment by kishsak** [comment]
hi is the dropdown issue fixed?

**Comment by Natsume-197** [comment]
I've encountered this issue as well, even in 2.2.0 version, not only with dropdowns but also with modals. As a temporary solution, reverting to version 1.9.0 fixed it.

**Comment by hasyim404** [comment]
> I've encountered this issue as well, even in 2.2.0 version (using Vue 3), not only with dropdowns but also with modals. As a temporary solution, reverting to version 1.9.0 fixed it.

thx bro, i'am stress about that, downgrade the version 2.1.0 to 1.9.0 is worked 👍

**Comment by kgnfth** [comment]
> This issue is the same as mentioned on #278, and is yet to be resolved.
> 
> Increasing the timeout does decrease the frequency of this issue, but it does still occasionally occur, it could be increased more to further improve it but the JS features are unresponsive for the duration of the timeout, which doesn't seem like a great solution to this.
> 
> Something has been introduced in v2 that is causing this issue as I've rolled back to 1.9.0 and this problem does not exist in that version.

Thx, this also works in vue3 now.

**Comment by kgnfth** [comment]
> I've encountered this issue as well, even in 2.2.0 version (using Vue 3), not only with dropdowns but also with modals. As a temporary solution, reverting to version 1.9.0 fixed it.

Thx for the fix

**Comment by jahaganiev** [comment]
Hey folks, since every framework has its ins and outs it's hard to know what exactly causing the issue. We suggest to open a new issue with Stackblitz demo and step by step details to reproduce the issue.

Here's a great [example](https://github.com/htmlstreamofficial/preline/issues/350#issuecomment-2139581724) how JS components should be integrated to specific frameworks.

Cheers!

**Comment by stnhnd** [comment]
```
"use client";

import { usePathname } from "next/navigation";
import { useEffect } from "react";

export default function PrelineScript() {
  const path = usePathname();

  useEffect(() => {
    const loadPreline = async () => {
      await import("preline/preline");

      window.HSStaticMethods.autoInit();
    };

    loadPreline();

    window.addEventListener("load", loadPreline);

    return () => {
      window.removeEventListener("load", loadPreline);
    };
  }, [path]);

  return null;
}
```

Hi, everybody, try this one, in my case its fix the problem.

**Comment by noobkisser** [comment]
This problm has caused me great suffering. any one resolved this yet? I might have to drop preline and move on to a different library

**Comment by jahaganiev** [comment]
> This problm has caused me great suffering. any one resolved this yet? I might have to drop preline and move on to a different library

You may open new issue with step by step details and our team will be more than happy to guide.
